### DISCUSSION TITLE
Potentially provide lightweight installation?

### DISCUSSION THREAD

**Discussion by williamz-homelight** [assafelovic/gpt-researcher#664] [General]
Hi there,  is it possible to do a lightweight install (like without hugging face support) now or planned in the future?  The docker image pretty unwieldy now.

**Comment by ElishaKay** [comment]
Hey @williamz-homelight,

It's a fair point.
Which dependencies do you think we should slice out for the slim version?

Also, [try this image](https://hub.docker.com/r/gptresearcher/gpt-researcher) and let me know if it's better for you.

**Reply by williamz-homelight** [reply]
I think carving out local llm / embedding model support is a good start? e.g. I tried building without the huggingface dep and resulted in image < 3g.

**Reply by ElishaKay** [reply]
sounds awesome @williamz-homelight 

I tend to agree!
A lightweight image would make testing new services much faster.

Is your only change in the requirements.txt?
In any case, send me over your changes either in this thread or in a PR - we can push up a lightweight image to dockerhub & update it as we go.

We have a big new release coming tomorrow, so hopefully this lightweight option can include those changes & be doable later this week if all goes well
